lib/pharos/phases/configure_dns.rb

```diff
@@ -45,52 +46,53 @@ def max_unavailable
 
       # @param replicas [Integer]
       # @param nodes [Integer]
-      def patch_kubedns(replicas:, max_surge:, max_unavailable:)
-        logger.info { "Patching kube-dns addon with #{replicas} replicas (max-surge #{max_surge}, max-unavailable #{max_unavailable})..." }
+      def patch_deployment(name, replicas:, max_surge:, max_unavailable:)
+        logger.info { "Patching #{name} deployment with #{replicas} replicas (max-surge #{max_surge}, max-unavailable #{max_unavailable})..." }
 
-        resource = Pharos::Kube.session(@master.api_address).resource(
-          apiVersion: 'extensions/v1beta1',
-          kind: 'Deployment',
-          metadata: {
-            namespace: 'kube-system',
-            name: 'kube-dns'
-          },
-          spec: {
-            replicas: replicas,
-            strategy: {
-              type: "RollingUpdate",
-              rollingUpdate: {
-                maxSurge: max_surge, # must be zero for a two-node cluster
-                maxUnavailable: max_unavailable, # must be at least one, even for a single-node cluster
-              }
-            },
-            template: {
-              spec: {
-                affinity: {
-                  podAntiAffinity: {
-                    requiredDuringSchedulingIgnoredDuringExecution: [
-                      {
-                        labelSelector: {
-                          matchExpressions: [
-                            {
-                              key: "k8s-app",
-                              operator: "In",
-                              values: [
-                                "kube-dns"
-                              ]
-                            }
-                          ]
-                        },
-                        topologyKey: "kubernetes.io/hostname"
-                      }
-                    ]
-                  }
+        Pharos::Kube.session(@master.api_address).resource_client('apps/v1').patch_deployment(
+          name,
+          {
+            spec: {
+              replicas: replicas,
+              strategy: {
+                type: "RollingUpdate",
+                rollingUpdate: {
+                  maxSurge: max_surge, # must be zero for a two-node cluster
+                  maxUnavailable: max_unavailable, # must be at least one, even for a single-node cluster
+                }
+              },
+              template: {
+                spec: {
+                  affinity: {
+                    podAntiAffinity: {
+                      requiredDuringSchedulingIgnoredDuringExecution: [
+                        {
+                          labelSelector: {
+                            matchExpressions: [
+                              {
+                                key: "k8s-app",
+                                operator: "In",
+                                values: [name]
```

Comment:
The `podAntiAffinity` is broken, the pods get scheduled on the same node.

The `coredns` deployment uses a `k8s-app: kube-dns` label, so this shouldn't be `coredns`.

Reply:
Fixed 

```
NAME                                          READY     STATUS    RESTARTS   AGE       IP               NODEt
coredns-dcb4c7ddd-pn96m                       1/1       Running   0          1m        10.32.1.8        terom-xenial-test
coredns-dcb4c7ddd-tk2dm                       1/1       Running   0          1m        10.32.2.29       terom-bionic-test
```